protoc-gen-openapiv2/internal/genopenapi/template_test.go

```diff
@@ -3918,6 +3918,7 @@ func TestTemplateToOpenAPIPath(t *testing.T) {
 		{"/{user.name=prefix/*}:customMethod", "/{user.name}:customMethod"},
 		{"/{user.name=prefix1/*/prefix2/*}:customMethod", "/{user.name}:customMethod"},
 		{"/{parent=prefix/*}/children:customMethod", "/{parent}/children:customMethod"},
+		{"/item/search:items/{item_no_query}", "/item/search:items/{item_no_query}"},
```

Comment:
it should be camel case here.. `"/item/search:items/{itemNoQuery}"`

Maybe you need to add a test to `TestTemplateWithJsonCamelCase` https://github.com/grpc-ecosystem/grpc-gateway/blob/main/protoc-gen-openapiv2/internal/genopenapi/template_test.go#L3843C6-L3843C35